akd_local_auditor/src/storage/s3/test.rs

```diff
@@ -0,0 +1,368 @@
+    // create the bucket
+    client
+        .create_bucket()
+        .bucket(bucket.to_string())
+        .send()
+        .await?;
+    // Generate a block of real, verifiable audit proofs
+    let proofs = generate_audit_proofs(n_blobs)
+        .await
+        .map_err(|akd_err| anyhow::anyhow!("AKD Error generating proofs: {}", akd_err))?;
+
+    // upload each proof blob into S3
+    for AuditInformation {
+        chash,
+        phash,
+        proof,
+    } in proofs
+    {
+        // Generate the s3 compat format
+        let blobs = akd::proto::generate_audit_blobs(vec![phash, chash], proof)
+            .map_err(|err| anyhow::anyhow!("Error generating audit blob {}", err))?;
+        // Grab the blob + upload it
+        if let Some(blob) = blobs.first() {
+            let byte_stream = ByteStream::from(blob.data.clone());
+            let name = blob.name.to_string();
+            client
+                .put_object()
+                .bucket(bucket.to_string())
+                .key(name)
+                .body(byte_stream)
+                .send()
+                .await?;
+        } else {
+            assert!(
+                false,
+                "We should never generate an empty blob array, but if we do crash hard & fast!"
+            );
+        }
+    }
+    Ok(())
+}
+
+fn get_s3_storage(bucket: &str) -> S3AuditStorage {
+    let clap_args = S3ClapSettings {
+        bucket: bucket.to_string(),
+        region: TEST_REGION.to_string(),
+        endpoint: Some(TEST_ENDPOINT.to_string()),
+        access_key: Some(ACCESS_KEY.to_string()),
+        secret_key: Some(SECRET_KEY.to_string()),
+    };
+    let storage: S3AuditStorage = (&clap_args).into();
+    storage
+}
+
+// =========================== Test Cases =========================== //
+
+#[tokio::test]
+#[ignore]
+async fn integration_test_bucket_listing() -> Result<()> {
+    // make sure we have a valid bucket name, that's "somewhat" unique
+    let bucket = bucket_name!();
+    log::debug!("Test bucket name is {}", bucket);
+    assert!(matches!(is_bucket_name_valid(&bucket), Ok(_)));
+
+    // Get the storage reader
+    let storage = get_s3_storage(&bucket);
+    let shared_config = storage.get_shared_config().await;
+
+    // Populate the test storage
+    populate_test_storage(&shared_config, &bucket, 10).await?;
+
+    // List the epochs found in the storage layer
+    let mut epoch_summaries: Vec<EpochSummary> = storage.list_proofs().await?;
+    epoch_summaries.sort_by(|a, b| a.name.epoch.cmp(&b.name.epoch));
+
+    // There should be 10 proofs in the storage layer
+    log::info!(
+        "There are {} epochs in the storage layer",
+        epoch_summaries.len()
+    );
+    assert_eq!(10, epoch_summaries.len());
+
+    // check the linear history of the proofs
+    log::info!("Checking linear history of audit proofs");
+    let mut i = 0u64;
+    for summary in epoch_summaries {
+        assert_eq!(i, summary.name.epoch);
+        i += 1;
+    }
```

Comment:
I believe you can use [`enumerate`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.enumerate) here to avoid having to maintain your own `i` variable.